[DOC REVIEW for PR #1640] HTTP trigger: add CORS attributes; Kafka trigger reference: fix typo (#1645)
nuclio/nuclio: docs/reference/triggers/http.md

| port | int | The NodePort (or equivalent) on which the function will serve HTTP requests. If empty, chooses a random port within the platform range. |
| ingresses.(name).host | string | The host to which the ingress maps. |
| ingresses.(name).paths | list of strings | The paths that the ingress handles. Variables of the form `{{.<NAME>}}` can be specified using `.Name`, `.Namespace`, and `.Version`. For example, `/{{.Namespace}}-{{.Name}}/{{.Version}}` will result in a default ingress of `/namespace-name/version`. |
| readBufferSize | int | Per-connection buffer size for reading requests. |

> | cors.enabled | bool | `true` to enable cross-origin resource sharing (CORS) for each function; (default: `false`). |

Reviewer: ```suggestion
| cors.enabled | bool | `true` to enable cross-origin resource sharing (CORS) for the function; (default: `false`). |
```
function <-> trigger  is 1:n so a trigger configuration refers to one function (multiple functions are not relevant in this context)

Author: I already removed "for each function" per Liran's suggestion — see the review comments and commit c4e882e.

| cors.allowOrigin | string | Indicates that the CORS response can be shared with requesting code from the specified origin (`Access-Control-Allow-Origin` response header); (default: `'*'` to allow sharing with any origin, for requests without credentials). |

> | cors.allowMethods | list of strings | HTTP methods that can be used when accessing the resource in response to a preflight request (`Access-Control-Allow-Methods` response header); (default: `"HEAD, GET, POST, PUT, DELETE, OPTIONS"`). |

Reviewer 2: ```suggestion
| cors.allowMethods | list of strings | The allowed HTTP methods that can be used when accessing the resource. (`Access-Control-Allow-Methods` response header); (default: `"HEAD, GET, POST, PUT, DELETE, OPTIONS"`). |
```

> | cors.allowHeaders | list of strings | HTTP headers can be used during the actual request (`Access-Control-Allow-Headers` response header); (default: `"Accept, Content-Length, Content-Type, X-nuclio-log-level"`). |

Reviewer 2: ```suggestion
| cors.allowHeaders | list of strings | The allowed HTTP headers that can be used when accessing the resource. (`Access-Control-Allow-Headers` response header); (default: `"Accept, Content-Length, Content-Type, X-nuclio-log-level"`). |
```

Author: "The allowed" and "that can be used" essentially say the same thing and it's redundant to use both. (Also, the "The" in this case is not required, specifically as we don't use full sentences, but I don't mind adding it.)
I based my `allowMethods` and `allowHeaders` descriptions on the MDN descriptions, which weren't phrased in a similar manner:
- `Access-Control-Allow-Methods` (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Methods) —
  > The `Access-Control-Allow-Methods` response header specifies the method or methods allowed when accessing the resource in response to a preflight request.
- `Access-Control-Allow-Headers` (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Headers) —
  > The `Access-Control-Allow-Headers` response header is used in response to a preflight request which includes the `Access-Control-Request-Headers` to indicate which HTTP headers can be used during the actual request.
As you say the description should be similar, how about the following?
```
| cors.allowMethods | list of strings | The HTTP methods that can be used when accessing the resource (`Access-Control-Allow-Methods` response header); (default: `"HEAD, GET, POST, PUT, DELETE, OPTIONS"`). |
| cors.allowHeaders | list of strings | The HTTP headers that can be used when accessing the resource (`Access-Control-Allow-Headers` response header); (default: `"Accept, Content-Length, Content-Type, X-nuclio-log-level"`). |
```
If you want to use "allowed" (even though i'ts mentioned in the response header names), we can use this (even though I would have preferred "permitted" to "allowed", but let's keep it close to the header names):
```
| cors.allowMethods | list of strings | The allowed HTTP methods, which can be used when accessing the resource (`Access-Control-Allow-Methods` response header); (default: `"HEAD, GET, POST, PUT, DELETE, OPTIONS"`). |
| cors.allowHeaders | list of strings | The allowed HTTP headers, which can be used when accessing the resource (`Access-Control-Allow-Headers` response header); (default: `"Accept, Content-Length, Content-Type, X-nuclio-log-level"`). |
```

Reviewer 2: the latter sounds good to me

Author: Changed — commit d570b1e.
